packages/fx-core/src/component/generator/generator.ts

```diff
@@ -204,6 +199,7 @@ export async function sampleDefaultOnActionError(
   error: Error
 ): Promise<void> {
   context.logProvider.error(error.message);
+  if (error instanceof BaseComponentInnerError) throw error.toFxError();
```

Comment:
Why do we need to add this logic here?

Reply:
The original process for both templates and samples would first try toFxError(), which doesn't work in the new template process, so this logic has been moved to the error handling for samples. Since there is no fallback for the sample's generation error, the BaseComponentInnerError that is thrown needs to execute toFxError() first.